packages/@aws-cdk/aws-redshift/lib/cluster.ts

```diff
@@ -0,0 +1,413 @@
+
+    /**
+     * Number of compute nodes in the cluster
+     *
+     * Value must be at least 1 and no more than 100.
+     *
+     * @default 1
+     */
+    readonly numberOfNodes?: number;
+
+    /**
+     * The node type to be provisioned for the cluster.
+     *
+     * @default NodeType.DC2_LARGE
+     */
+    readonly nodeType?: NodeType;
+
+    /**
+     * Settings for the individual instances that are launched
+     *
+     * @default ClusterType.MULTI_NODE
+     */
+    readonly clusterType?: ClusterType;
+
+    /**
+     * What port to listen on
+     *
+     * @default - The default for the engine is used.
+     */
+    readonly port?: number;
+
+    /**
+     * Whether to enable encryption of data in the cluster
+     *
+     * @default false
+     */
+    readonly encrypted?: boolean
+
+    /**
+     * The KMS key for storage encryption.
+     * will be set to `true`.
+     *
+     * @default - default master key.
+     */
+    readonly kmsKey?: kms.IKey;
+
+    /**
+     * A preferred maintenance window day/time range. Should be specified as a range ddd:hh24:mi-ddd:hh24:mi (24H Clock UTC).
+     *
+     * Example: 'Sun:23:45-Mon:00:15'
+     *
+     * @default - 30-minute window selected at random from an 8-hour block of time for
+     * each AWS Region, occurring on a random day of the week.
+     * @see https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/USER_UpgradeDBInstance.Maintenance.html#Concepts.DBMaintenance
+     */
+    readonly preferredMaintenanceWindow?: string;
+
+    /**
+     * The VPC to place the cluster in.
+     *
+     * @default a new Vpc will be created
+     */
+    readonly vpc?: ec2.IVpc;
+
+    /**
+     * Security group.
+     *
+     * @default a new security group is created.
+     */
+    readonly securityGroups?: ec2.ISecurityGroup[];
+
+    /**
+     * Username and password for the administrative user
+     */
+    readonly masterUser: Login;
+
+    /**
+     * A list of AWS Identity and Access Management (IAM) roles that can be used by the cluster to access other AWS services.
+     * Specify a maximum of 10 roles.
+     *
+     * @default - A role is automatically created for you
+     */
+    readonly iamRoles?: IRole[];
+
+    /**
+     * Name of a database which is automatically created inside the cluster
+     *
+     * @default - Database is not created in cluster.
+     */
+    readonly defaultDatabaseName?: string;
+
+    /**
+     * Specifies logging information, such as queries and connection attempts, for the specified Amazon Redshift cluster.
+     *
+     * @default - No Logs
+     */
+    readonly loggingProperties?: LoggingProperties;
+
+    /**
+     * The removal policy to apply when the cluster and its instances are removed
+     * from the stack or replaced during an update.
+     *
+     * @default - Retain cluster.
+     */
+    readonly removalPolicy?: RemovalPolicy
+}
+
+/**
+ * Properties for Redshift Logging
+ */
+export interface LoggingProperties {
```

Comment:
I don't like this interface. Can we instead put these into `ClusterProps`, called `loggingBucket?: IBucket` and `loggingKeyPrefix?: string`?